### PR TITLE
Integrate MLflow; allow multiple sweeps

### PR DESCRIPTION
**Pull request by Nking92**
This PR integrates MLflow support into the backend. MLflow can be controlled from the CompatibilityAnalysis API with parameters use_ml_flow and ml_flow_run_name. If use_ml_flow=False, we will not log the sweep with MLflow.

This PR also contains bug fixes for https://github.com/microsoft/BackwardCompatibilityML/issues/87. The widget can now be instantiated multiple times, and a sweep can be run multiple times without crashing the widget.

### CODE REVIEW

File: backwardcompatibilityml/sweep_management.py [hunk 1]
```diff
@@ -76,7 +81,9 @@ def __init__(self, folder_name, number_of_epochs, h1, h2, training_set, test_set
                  performance_metric=model_accuracy,
                  get_instance_image_by_id=None,
                  get_instance_metadata=None,
-                 device="cpu"):
+                 device="cpu",
+                 use_ml_flow=True,
```
Comment (ilmarinen) [inline backwardcompatibilityml/sweep_management.py:85]: Let's set this to default `False`.

File: backwardcompatibilityml/helpers/training.py [hunk 2]
```diff
@@ -719,7 +720,9 @@ def compatibility_sweep(sweeps_folder_path, number_of_epochs, h1, h2,
                         new_error_loss_kwargs=None,
                         strict_imitation_loss_kwargs=None,
                         get_instance_metadata=None,
-                        device="cpu"):
+                        device="cpu",
+                        use_ml_flow=True,
```
Comment (ilmarinen) [inline backwardcompatibilityml/helpers/training.py:724]: Let's set this to default `False`.

File: backwardcompatibilityml/widget/compatibility_analysis.py [hunk 3]
```diff
@@ -219,7 +223,9 @@ def __init__(self, folder_name, number_of_epochs, h1, h2, training_set, test_set
                  strict_imitation_loss_kwargs=None,
                  get_instance_image_by_id=None,
                  get_instance_metadata=None,
-                 device="cpu"):
+                 device="cpu",
+                 use_ml_flow=True,
```
Comment (ilmarinen) [inline backwardcompatibilityml/widget/compatibility_analysis.py:227]: Let's set this to default `False`.

**Comment by ilmarinen** [comment]
Looks good. Two minor changes:

1. Let's have the default behaviour be for it to not use MLFlow, i.e. let it default to using our existing methods for running the sweeps.

2. Add mlflow (with pinned version number) to the requirements.txt.

**Review by ilmarinen — APPROVED** [review]
@Nking92 +1

**Review by jlema — APPROVED** [review]

**Review by nushib — APPROVED** [review]

**Pull request merged**